### PR TITLE
Document the different combinations of build outputs that Emscripten can produce.

### PR DESCRIPTION
**Pull request by juj**
Document the different combinations of build outputs that Emscripten can produce.

### CODE REVIEW

File: site/source/docs/compiling/Building-Projects.rst [hunk 1]
```diff
@@ -53,6 +53,29 @@ To build with Emscripten, you would instead use the following commands:
...
 
+.. _building-projects-build-outputs:
+
+Emscripten build output files
+=============================
+
+Emscripten compiler output often consists of several files and not just one. The set of produced files changes depending on the final linker flags passed to `emcc/em++`. Here is a cheat sheet of which files are produced under which conditions:
+
+ - `emcc ... -o output.html` builds a `output.html` file as an output, as well as an accompanying `output.js` launcher file, and a `output.wasm` WebAssembly file.
+ - `emcc ... -o output.js` omits generating a HTML launcher file (expecting you to provide it yourself), and produces two files, `output.js` and `output.wasm`.
```
Comment (kripken) [inline site/source/docs/compiling/Building-Projects.rst]: I'd rephrase this as: we just emit js and wasm here. If you run in node.js, that might be enough, or you can write your own HTML to load it in etc.

File: site/source/docs/compiling/Building-Projects.rst [hunk 2]
```diff
@@ -53,6 +53,29 @@ To build with Emscripten, you would instead use the following commands:
...
+ - `emcc ... -o output.html -s WASM=0 -s PRECISE_F32=2` (combination of targeting .html, asm.js and PRECISE_F32=2) implies as if `--separate-asm` was passed, so also produces `output.asm.js`.
+ - `emcc ... -o output.{html,js} --emit-symbol-map` produces a file `output.{html,js}.symbols` if WebAssembly is being targeted (`-s WASM=0` not specified), or if asm.js is being targeted and `-Os`, `-Oz` or `-O2` or higher is specified, but debug level setting is `-g1` or lower (i.e. if symbols minification did occur).
+ - `emcc ... -o output.{html,js} -s WASM=0 --memory-init-file 1` causes the generation of `output.{html,js}.mem` memory initializer file. Pasing `-O2`, `-Os` or `-Oz` also implies `--memory-init-file 1`.
+ - `emcc ... -o output.{html,js} -g4` generates a source map file `output.wasm.map`. If targeting asm.js with `-s WASM=0`, the filename is `output.{html,js}.map`.
+ - `emcc ... -o output.{html,js} --preload-file xxx` directive generates a preloaded MEMFS filesystem file `output.data`.
+ - `emcc ... -o output.{html,js} --preload-file xxx` directive generates a preloaded MEMFS filesystem file `output.data`.
```
Comment (kripken) [inline site/source/docs/compiling/Building-Projects.rst]: this line appears twice

File: site/source/docs/compiling/Building-Projects.rst (hunk 1 above)
Comment (juj) [reply site/source/docs/compiling/Building-Projects.rst]: ok, updated.

**Comment by juj** [comment]
> Maybe another way to write this is, instead of specifying complete commands on each line, just the influence of individual flags. So one line in this could could mention `-s WASM=0`, one line could mention `--separate-asm`, etc. - thoughts?

I tried to be as orthogonal on these as possible. For example I cannot document `--separate-asm` without `-s WASM=0`, because that is a build error.

The fact for example that I list `emcc ... -o output.{html,js} -g4` instead of `emcc ... -g4` for source maps is because I want to concisely say that source maps will only appear when building to html or js (and not e.g. when building to .bc or .o).

The reason btw I am documenting these is that a trend I am seeing is that build systems tend to manually configure the expected build output files for build dependency tracking for what to rebuild, and for post-build copy deploy steps to a final output directory. It seems to be a common source of bugs where a build system does not take into account that some file was generated, or that they assume something was generated even though it did not.

**Comment by kripken** [comment]
Makes sense, lgtm.

Btw, when creating and when landing a PR like this, please use `[ci skip]` in the title of the commit or PR, to save CI resources (e.g. right now we are very backlogged there).

**Comment by juj** [comment]
Thanks, I'll remember that flag, that's useful.
